src/js/components/CheckBox/CheckBox.js

```diff
@@ -191,7 +192,7 @@ const CheckBox = forwardRef(
             if (onChange) onChange(event);
           }}
         />
-        {visual}
+        {children ? children({ checked, indeterminate }) : <>{visual}</>}
```

Comment:
Why is there a Fragment around `visual`?
```suggestion
        {children ? children({ checked, indeterminate }) : visual}
```